crates/vm/levm/src/precompiles.rs

```diff
@@ -887,10 +892,113 @@ pub fn blake2f(
     Ok(Bytes::from(output))
 }
 
+// Taken from the same name function from crates/common/types/blobs_bundle.rs
+fn kzg_commitment_to_versioned_hash(data: &[u8; 48]) -> H256 {
+    use k256::sha2::Digest;
+    let mut versioned_hash: [u8; 32] = k256::sha2::Sha256::digest(data).into();
+    versioned_hash[0] = VERSIONED_HASH_VERSION_KZG;
+    versioned_hash.into()
+}
+
+fn verify_kzg_proof(
+    commitment_bytes: &[u8; 48],
+    x: &[u8; 32],
+    y: &[u8; 32],
+    proof_bytes: &[u8; 48],
+) -> Result<bool, VMError> {
+    let commitment_bytes =
+        Bytes48::from_slice(commitment_bytes).map_err(|_| PrecompileError::EvaluationError)?; // Could be ParsingInputError
+    let z_bytes = Bytes32::from_slice(x).map_err(|_| PrecompileError::EvaluationError)?;
+    let y_bytes = Bytes32::from_slice(y).map_err(|_| PrecompileError::EvaluationError)?;
+    let proof_bytes =
+        Bytes48::from_slice(proof_bytes).map_err(|_| PrecompileError::EvaluationError)?;
+
+    let settings =
+        KzgSettings::load_trusted_setup_file().map_err(|_| PrecompileError::EvaluationError)?;
+
+    kzg_rs::kzg_proof::KzgProof::verify_kzg_proof(
+        &commitment_bytes,
+        &z_bytes,
+        &y_bytes,
+        &proof_bytes,
+        &settings,
+    )
+    .map_err(|_| VMError::PrecompileError(PrecompileError::EvaluationError))
+}
+
 fn point_evaluation(
-    _calldata: &Bytes,
-    _gas_for_call: u64,
-    _consumed_gas: &mut u64,
+    calldata: &Bytes,
+    gas_for_call: u64,
+    consumed_gas: &mut u64,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    if calldata.len() != 192 {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    // Consume gas
+    let gas_cost = 50000;
+    increase_precompile_consumed_gas(gas_for_call, gas_cost, consumed_gas)?;
+
+    // Parse inputs
+    let hash = calldata
+        .get(..32)
+        .ok_or(PrecompileError::ParsingInputError)?;
```

Comment:
Does this translate to an internal error? Because if not maybe we should make it internal because this shouldn't ever happen, right?

Reply:
Done in [Change versioned_hash parsing](https://github.com/lambdaclass/ethrex/pull/1569/commits/0a89f32452096a5d3871e491528c34f5aadc85f4)